fhir-server-test/src/test/java/com/ibm/fhir/server/test/operation/ReindexOperationTest.java

```diff
@@ -457,29 +463,60 @@ public void testReindexWithPatientResourceType() {
         assertEquals(r.getStatus(), Status.OK.getStatusCode());
     }
 
+    /**
+     * Wrapper for strings.
+     * @param str the string
+     * @return the string
+     */
+    protected static com.ibm.fhir.model.type.String str(String str) {
+        return com.ibm.fhir.model.type.String.of(str);
+    }
+
+    /**
+     * Wrapper for integers.
+     * @param val the integer
+     * @return the integer
+     */
+    protected static com.ibm.fhir.model.type.Integer intValue(int val) {
+        return com.ibm.fhir.model.type.Integer.of(val);
+    }
+
     @Test
     public void testReindex_indexIds() {
-        List<Parameter> parameters = new ArrayList<>();
-        parameters.add(
-            Parameter.builder()
-                .name(string("indexIds"))
-                .value(string("2,4,6,8,10"))
-                .build());
-
-        Parameters.Builder builder = Parameters.builder();
+        // Allocation of logical_resource_id values (from fhir_sequence) is more complicated
+        // now, so we can't depend on valid values being 2,4,6,8,10 etc. Need to perform a
+        // $retrieve-index operation first
+        final String reindexTimestamp = Instant.now().toString(); // ISO UTC
+        Builder builder = Parameters.builder();
+        builder.parameter(Parameter.builder().name(str(COUNT_PARAM)).value(intValue(MAX_RETRIEVE_COUNT)).build());
+        builder.parameter(Parameter.builder().name(str(NOT_MODIFIED_AFTER_PARAM)).value(str(reindexTimestamp)).build());
         builder.id(UUID.randomUUID().toString());
-        builder.parameter(parameters);
-        Parameters ps = builder.build();
-
-        Entity<Parameters> entity = Entity.entity(ps, FHIRMediaType.APPLICATION_FHIR_JSON);
+        Parameters parameters = builder.build();
+        
 
+        Entity<Parameters> requestEntity = Entity.entity(parameters, FHIRMediaType.APPLICATION_FHIR_JSON);
         Response r = getWebTarget()
+                .path("/$retrieve-index")
+                .request(FHIRMediaType.APPLICATION_FHIR_JSON)
+                .header("X-FHIR-TENANT-ID", "default")
+                .header("X-FHIR-DSID", "default")
+                .post(requestEntity, Response.class);
+
+        assertEquals(r.getStatus(), Status.OK.getStatusCode());
+        
+        // We can turn around and use the Parameters response directly in our reindex call
+        Parameters indexIdParams = r.readEntity(Parameters.class);
+        assertTrue(indexIdParams.getParameter().size() == 1);
```

Comment:
this assertion is failing.  is it needed?

Reply:
I made the response handling more robust, but more importantly, changed setup() to actually inject some data into the database before we try to reindex anything.